ui/pages/confirmations/confirm/confirm.tsx

```diff
@@ -25,12 +24,16 @@ const Confirm = () => {
   syncConfirmPath();
   setConfirmationAlerts();
   const processAction = useConfirmationAlertActions();
+  const [showAdvancedDetails, setShowAdvancedDetails] = React.useState(false);
 
   return (
     <AlertActionHandlerProvider onProcessAction={processAction}>
       <Page className="confirm_wrapper">
         <Nav />
-        <Header />
+        <Header
+          showAdvancedDetails={showAdvancedDetails}
+          setShowAdvancedDetails={setShowAdvancedDetails}
```

Comment:
Currently we're passing these properties into the `Header` and then `HeaderInfo`, plus `Info`.

Assuming this advanced toggle will be a confirmation wide concept (which sounds great), is it worth moving it to a React context, all encapsulated within a hook like `useAdvancedSettingsContext`?

That way any component anywhere in the tree can vary based on whether we're in the advanced view?

Reply:
I appreciate the suggestion but tbh I would prefer keeping it this way for the moment if that's ok. I don't think advanced settings will be used in enough different components to warrant this premature optimisation, at least not yet. This would require more boilerplate and adding to the number of contexts, which I'm not certain is better than adding one prop.